pints/_optimisers/_hager_zhang_line_search.py

```diff
@@ -0,0 +1,899 @@
+#
+# Fixed learning-rate gradient descent.
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+
+import pints
+import numpy as np
+from numpy.linalg import norm
+
+
+class HagerZhang(pints.Optimiser):
+    """
+    Gradient-descent method with a fixed learning rate.
+    """
+
+    def __init__(self, x0, sigma0=None, boundaries=None):
+        super(HagerZhang, self).__init__(x0, sigma0, boundaries)
+
+        # Set optimiser state
```

Comment:
I take it this method only works for 1d problems. Should there be a check in the constructor for this? Or in a LineSearch class that this one extends?